src/Console/Arguments.php

```diff
@@ -74,9 +75,9 @@ public function getArguments(): array
      * Get positional arguments by index.
      *
      * @param int $index The argument index to access.
-     * @return string|null The argument value or null
+     * @return array|string|null The argument value or null
      */
-    public function getArgumentAt(int $index): ?string
+    public function getArgumentAt(int $index): array|string|null
```

Comment:
To not make this more problematic in the end use (consumer side - see also phpstan fails) - I wonder if this should also be clear about scalar vs array in usage:

    public function getArgumentAt(int $index): string|null
    public function getArrayArgumentAt(int $index): array

There could even be a third one if one really wants to use the "mixed" one and manually is_array/is_scalar on it:

    public function getArgumentAt(int $index): array|string|null
    public function getArrayArgumentAt(int $index): array
    public function getStringArgumentAt(int $index): string|null

Reply:
don't know what methodd will return or a too more complex output could be hard to code/maintain.

But one method by type is not obvious ... 
I think i prefer to know what is the output of a method, one method return array or null, another string or null, easiest and more clear code i think.

Reply:
The option or argument is always one, just the value is multiple times aka array.
The other method is getBool.. it only makes sense to follow more that pattern.

GetMultipleOption
Get MultipleArgument
Sounds a bit wrong in that context no?

But my Main point is that array|scalar is always a Bad return type. Requiring more checks on user side.

Reply:
`getArgumentAt()` could return the unparsed value. That would let you get the scalar type 🤷